### DISCUSSION TITLE
expose config doesn't work in unit test mode

### DISCUSSION THREAD

**Discussion by diman82** [apache/airflow#24287] [General]
When spinning a local airflow webserver while in unit test mode and exposing config
```
  export AIRFLOW__CORE__UNIT_TEST_MODE=True
  export AIRFLOW__WEBSERVER__EXPOSE_CONFIG=True
```
I get an error:
```
    with open(AIRFLOW_CONFIG) as file:
FileNotFoundError: [Errno 2] No such file or directory: '/mnt/d/dev/data-management-airflow-2/tests/airflow.cfg'
```

Seems like airflow is ignoring UNIT_TEST_MODE (unittests.cfg) and fallbacks to default airflow.cfg file..

**Comment by potiuk** [comment]
I think UNIT_TEST_MODE only works for running tests not for Airlfow webserver. It makes very little sense to open "real" airflow components with unit tests mode. It is foreseen to .... welll .... run unit tests.

**Reply by diman82** [reply]
You definitely got a point here.
And yet, I'm still using a config file (unittests.cfg), so it does makes sense to me to be able to inspect the configuration values..

**Reply by potiuk** [reply]
Why not airflow.cfg? What's wrong with it?

**Reply by diman82** [reply]
because I would like to separate configuration options for testing and normal execution..

**Reply by potiuk** [reply]
Then use different folder for testing and set AIRFLOW_HOME to it.

**Reply by potiuk** [reply]
Really using unittest.cfg for webserver config makes absolutely no sense whatsoever.
